lib/engine/game/g_18_christmas_eve/step/buy_sell_par_gift_shares.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/buy_sell_par_shares'
+
+module Engine
+  module Game
+    module G18ChristmasEve
+      module Step
+        class BuySellParGiftShares < Engine::Step::BuySellParShares
+          def description
+            'Sell/Buy/Sell Shares and/or Gift Certs'
+          end
+
+          def round_state
+            super.merge!(
+              {
+                presidencies_gifted: [],
+              }
+            )
+          end
+
+          def actions(entity)
+            return [] unless entity == current_entity
+
+            actions = super
+            actions << 'choose' if can_gift_any?
+            actions << 'pass' unless actions.empty? && actions.include?('pass')
+            actions
+          end
+
+          def choice_available?(entity)
+            return false unless @game.sellable_turn? && entity.corporation?
+
+            @choices = gift_choices(entity)
+            @choices != {}
+          end
+
+          def gift_choices(entity)
+            return {} unless entity&.corporation? && entity&.president?(current_entity)
+
+            choices = {}
+            # Cert only
+            # If we own 20%, then we've only got the pres cert to gift
+            if current_entity.percent_of(entity) != 20
+              choices.merge!(@game.players
+                .map.with_index { |p, i| [p, i] }
+                .reject { |p, _i| p == current_entity }
+                .select { |p, _i| p.percent_of(entity).zero? }
+                .to_h { |p, i| ["cert_#{i}_#{entity.id}", "Cert to #{p.name}"] })
+            end
+
+            # We can only give the pres cert if it would mean that player has most/equal most shares
+            president_possible = current_entity.percent_of(entity) <= 40 &&
+              @game.players.reject { |p| p == current_entity }.none? { |p| p.percent_of(entity) > 20 }
+            if president_possible
+              choices.merge!(@game.players
+                .map.with_index { |p, i| [p, i] }
+                .reject { |p, _i| p == current_entity }
+                # Co prez can't be gifted twice in a round
+                .reject { |_p, _i| @round.presidencies_gifted&.include? entity }
+                .select { |p, _i| p.percent_of(entity).zero? }
+                .to_h { |p, i| ["prez_#{i}_#{entity.id}", "Presidency to #{p.name}"] })
+            end
+
+            choices
+          end
+
+          def can_gift_any?
+            @game.sellable_turn? && @game.corporations.any? { |e| gift_choices(e).any? }
```

Comment:
don't use any? for gift_choices, use empty?

Reply:
This because `!entity_choices(e).empty?` - I hope that's what you mean